src/modules/hip/kernel/resize.hpp

```diff
@@ -0,0 +1,735 @@
+
+    uint2 dstDimsWH;
+    dstDimsWH.x = dstImgSize[id_z].width;
+    dstDimsWH.y = dstImgSize[id_z].height;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
+    {
+        return;
+    }
+
+    int4 srcRoi_i4 = *(int4 *)&roiTensorPtrSrc[id_z];
+    uint2 srcDimsWH;
+    srcDimsWH.x = srcRoi_i4.z - srcRoi_i4.x + 1;
+    srcDimsWH.y = srcRoi_i4.w - srcRoi_i4.y + 1;
+    int widthLimit = srcRoi_i4.z;
+    int heightLimit = srcRoi_i4.w;
+    float wRatio = (float)srcDimsWH.x / (float)dstDimsWH.x;
+    float hRatio = (float)srcDimsWH.y / (float)dstDimsWH.y;
+    float hScale = 1.0f, wScale = 1.0f, hRadius = 1.0f, wRadius = 1.0f;
+
+    rpp_hip_compute_interpolation_scale_and_radius(interpolationType, srcDimsWH.x, dstDimsWH.x, &wScale, &wRadius, wRatio);
+    rpp_hip_compute_interpolation_scale_and_radius(interpolationType, srcDimsWH.y, dstDimsWH.y, &hScale, &hRadius, hRatio);
+    float wOffset = (wRatio - 1) * 0.5f - wRadius;
+    float hOffset = (hRatio - 1) * 0.5f - hRadius;
+    int wKernelSize = ceilf(wRadius * 2);
+    int hKernelSize = ceilf(hRadius * 2);
+
+    float rowWeight, colWeight, rowCoeff, colCoeff;
+    int srcLocationRowFloor, srcLocationColumnFloor;
+    resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.x, id_x, wRatio, widthLimit, &srcLocationColumnFloor, &colWeight, wOffset, 1);
+    resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.y, id_y, hRatio, heightLimit, &srcLocationRowFloor, &rowWeight, hOffset, 1);
+
+    T *srcPtrTemp[3];
+    srcPtrTemp[0] = srcPtr + (id_z * srcStridesNCH.x);
+    srcPtrTemp[1] = srcPtrTemp[0] + srcStridesNCH.y;
+    srcPtrTemp[2] = srcPtrTemp[1] + srcStridesNCH.y;
+
+    T *srcRowPtrsForInterp[3];
+    float3 outPixel_f3 = (float3)0.0f;
+    float rowCoeffSum = 0.0f, colCoeffSum = 0.0f;
+    for(int j = 0; j < hKernelSize; j++)
+    {
+        int rowIndex = fminf(fmaxf((int)(srcLocationRowFloor + j), 0), heightLimit);
+        srcRowPtrsForInterp[0] = srcPtrTemp[0] + rowIndex * srcStridesNCH.z;
+        srcRowPtrsForInterp[1] = srcPtrTemp[1] + rowIndex * srcStridesNCH.z;
+        srcRowPtrsForInterp[2] = srcPtrTemp[2] + rowIndex * srcStridesNCH.z;
+        rpp_hip_compute_interpolation_coefficient(interpolationType, (rowWeight - hRadius + j) * hScale , &rowCoeff);
+        rowCoeffSum += rowCoeff;
+
+        colCoeffSum = 0;
+        for(int k = 0; k < wKernelSize; k++)
+        {
+            int colIndex = fminf(fmaxf((int)(srcLocationColumnFloor + k), 0), widthLimit);
+            rpp_hip_compute_interpolation_coefficient(interpolationType, (colWeight - wRadius + k) * wScale , &colCoeff);
+            colCoeffSum += colCoeff;
+            float3 coeff_f3 = (float3)(colCoeff * rowCoeff);
+            outPixel_f3 += (make_float3(srcRowPtrsForInterp[0][colIndex], srcRowPtrsForInterp[1][colIndex], srcRowPtrsForInterp[2][colIndex]) * coeff_f3);
+        }
+    }
+    outPixel_f3 *= (float3)(1 / (rowCoeffSum * colCoeffSum));
```

Comment:
check if there is a possibility of division by zero here

Reply:
@rrawther 
There is a possibility of 0 as some interpolation functions will return 0 as coeff based on the weights being passed as input to interpolation functions
https://github.com/r-abishek/rpp/blob/ar/opt_resize_hip/src/include/hip/rpp_hip_common.hpp#L1750
https://github.com/r-abishek/rpp/blob/ar/opt_resize_hip/src/include/hip/rpp_hip_common.hpp#L1739

if we initialize the value to  very small number !=0  then 1/(very small number) will become a large value which when multiplied by outPixel_f3 makes it as a large number / can make it out of range

Is it fine if we can add two branch statements like below to handle this case?
`rowCoeffSum = (rowCoeffSum == 0.0f) ? 1.0f : rowCoeffSum;
    colCoeffSum = (colCoeffSum == 0.0f) ? 1.0f : colCoeffSum;
    outPixel_f3 *= (float3)(1 / (rowCoeffSum * colCoeffSum));`